controllers/metal3labelsync_controller_test.go

```diff
@@ -329,6 +334,537 @@ var _ = Describe("Metal3LabelSync controller", func() {
+				machine: &capi.Machine{
+					TypeMeta: metav1.TypeMeta{
+						Kind:       "Machine",
+						APIVersion: capi.GroupVersion.String(),
+					},
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      machineName,
+						Namespace: namespaceName,
+						Labels: map[string]string{
+							capi.ClusterLabelName: clusterName,
+						},
+					},
+				},
+				expectRequeue: true,
+			}),
+			Entry("Error fetching cluster", testCaseReconcile{
+				host: &bmh.BareMetalHost{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "testHost",
+						Namespace: namespaceName,
+						Labels: map[string]string{
+							"foo.metal3.io/bar": "blue",
+						},
+					},
+					Spec: bmh.BareMetalHostSpec{
+						ConsumerRef: &corev1.ObjectReference{
+							Name:       metal3machineName,
+							Namespace:  namespaceName,
+							Kind:       "Metal3Machine",
+							APIVersion: capm3.GroupVersion.String(),
+						},
+					},
+				},
+				metal3Machine: newMetal3Machine(metal3machineName, m3mObjectMetaWithOwnerRef(), nil, nil, false),
+				machine: &capi.Machine{
+					TypeMeta: metav1.TypeMeta{
+						Kind:       "Machine",
+						APIVersion: capi.GroupVersion.String(),
+					},
+					Status: capi.MachineStatus{
+						NodeRef: &corev1.ObjectReference{
+							Kind: "Node",
+							Name: "testNode",
+						},
+					},
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      machineName,
+						Namespace: namespaceName,
+						Labels: map[string]string{
+							capi.ClusterLabelName: clusterName,
+						},
+					},
+				},
+				expectError:   true,
+				expectRequeue: true,
+			}),
+			Entry("Error fetching Metal3Cluster", testCaseReconcile{
+				host: &bmh.BareMetalHost{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "testHost",
+						Namespace: namespaceName,
+						Labels: map[string]string{
+							"foo.metal3.io/bar": "blue",
+						},
+					},
+					Spec: bmh.BareMetalHostSpec{
+						ConsumerRef: &corev1.ObjectReference{
+							Name:       metal3machineName,
+							Namespace:  namespaceName,
+							Kind:       "Metal3Machine",
+							APIVersion: capm3.GroupVersion.String(),
+						},
+					},
+				},
+				metal3Machine: newMetal3Machine(metal3machineName, m3mObjectMetaWithOwnerRef(), nil, nil, false),
+				machine: &capi.Machine{
+					TypeMeta: metav1.TypeMeta{
+						Kind:       "Machine",
+						APIVersion: capi.GroupVersion.String(),
+					},
+					Status: capi.MachineStatus{
+						NodeRef: &corev1.ObjectReference{
+							Kind: "Node",
+							Name: "testNode",
+						},
+					},
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      machineName,
+						Namespace: namespaceName,
+						Labels: map[string]string{
+							capi.ClusterLabelName: clusterName,
+						},
+					},
+				},
+				cluster:       newCluster(clusterName, nil, nil),
+				expectError:   true,
+				expectRequeue: true,
+			}),
+			Entry("Cluster is paused", testCaseReconcile{
```

Comment:
Please correct me if I'm wrong, but this test and all the tests above it are not really specific to the label-sync controller. Is there any reason why they are necessary?

Reply:
I am trying to cover all the cases that might hit label-sync reconcile, so any changes in that function should not break the test. As an example removing 
```
	if annotations.IsPaused(cluster, metal3Cluster) {
		controllerLog.Info("Cluster and/or Metal3Cluster are currently paused. Remove pause to continue reconciliation.")
		return ctrl.Result{RequeueAfter: bmhSyncInterval}, nil
	}
```
from lable-sync reconcile should be detected by this `Entry("Cluster is paused"` is not that needed for our unit test ?